python-package/lightgbm/basic.py

```diff
@@ -80,10 +80,10 @@ def list_to_1d_numpy(data, dtype=np.float32, name='list'):
     elif isinstance(data, Series):
         if _get_bad_pandas_dtypes([data.dtypes]):
             raise ValueError('Series.dtypes must be int, float or bool')
-        if hasattr(data.values, 'values'):  # SparseArray
-            return data.values.values.astype(dtype)
+        if data.dtype == np.float32 or data.dtype == np.float64:
+            return np.array(data, dtype=data.dtype, copy=False)
```

Comment:
@guolinke According to this note
https://github.com/microsoft/LightGBM/blob/15566421bcc96d46bac353dee17b981a12827604/include/LightGBM/c_api.h#L310-L315
it seems that we cannot allow these memory savings, right?

Reply:
sorry for missing this comment.
yes, the `field` in dataset don't allow arbitrary data types.

Reply:
I think I could have a parameter, maybe called `new_type` with default value `None`. And when it is not none, convert to the new type if it is not that type.

Reply:
@guolinke Seems we always need to force the type, except training/test data, but it's not going through this function. So, I can simply remove that `if` statement.

Reply:
ping @guolinke